src/io/io_entry.c

```diff
@@ -363,26 +364,34 @@ int io_entry_write_fields(const char *state, const char *path,
         fprintf(stream, "\"sha512\":\"%s\",", buf);
     }
 
+    /*
+     * up till this point we have been appending commas to the end of the
+     * entries, this is because we don't know which hashes are going to be
+     * provided. Now that pathmd5 has been put on the stream we switch to
+     * commas before any new entries, allowing errors to simply exit and not
+     * output malformed json.
+     */
+
     unpack(buf, pathmd5, MD5_DIGEST_LENGTH);
-    fprintf(stream, "\"pathmd5\":\"%s\",", buf);
+    fprintf(stream, "\"pathmd5\":\"%s\"", buf); /* no commas before or after */
```

Comment:
I realize that the json is borked in some places but please include these types of changes into their own patch. Including them here along with the larger symlink changes makes it harder to test and even revert if need be.